File: Code/Chapter 7/C#/Exercise 7.3/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.Data.SqlClient;
#endregion

namespace Exercise_7_3
{
   partial class Form1 : Form
   {
      private DataSet myData;

      public Form1()
      {
         InitializeComponent();
         myData = new DataSet();
      }

      private void buttonUserData_Click(object sender, EventArgs e)
      {
         // If there is a data source, remove it.
         datagridView.DataSource = "";
         // Never hard code connection strings.
         // Usually you would get this from a config file
         string connectionString =
         "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;";

         using (SqlConnection testConnection = new SqlConnection(connectionString))
         {
            SqlCommand testCommand = testConnection.CreateCommand();
            testCommand.CommandText = "Select * from userTable";
            SqlDataAdapter dataAdapter = new SqlDataAdapter(testCommand);

            dataAdapter.Fill(myData, "UserTable");
         } // testConnection.Dispose called automatically.
         UpdateComboBox();
      }

      private void btnPermData_Click(object sender, EventArgs e)
      {
         // Never hard code connection strings.
         // Usually you would get this from a config file
         string connectionString =
         "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;";

         using (SqlConnection testConnection = new SqlConnection(connectionString))
         {
            SqlCommand testCommand = testConnection.CreateCommand();
            testCommand.CommandText = "Select PermissionType from PermissionsTable";
            SqlDataAdapter dataAdapter = new SqlDataAdapter(testCommand);

            dataAdapter.Fill(myData, "PermissionsTable");
         } // testConnection.Dispose called automatically.
         UpdateComboBox();
      }

      private void UpdateComboBox()
      {
         comboTables.Items.Clear();
         foreach (DataTable tbl in myData.Tables)
         {
            comboTables.Items.Add(tbl.TableName);
         }
      }

      private void cmbTables_SelectedIndexChanged(object sender, EventArgs e)
      {
         datagridView.DataSource = myData.Tables[comboTables.SelectedIndex];
      }
   }
}

File: Code/Chapter 7/VB.NET/Exercise 7.3/Form1.vb
Imports System.Data
Imports System.Data.SqlClient

Public Class Form1
    Private myData As New DataSet
   Private Sub buttonUserData_Click( _
      ByVal sender As System.Object, ByVal e As System.EventArgs) _
      Handles buttonUserData.Click
      ' If there is a data source, remove it.
      datagridView.DataSource = Nothing
      ' Never hard code connection strings.
      ' Usually you would get this from a config file
      Dim connectionString As String = _
       "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;"

      Using testConnection As SqlConnection = New SqlConnection(connectionString)
         Dim testCommand As SqlCommand = testConnection.CreateCommand()
         testCommand.CommandText = "Select * from userTable"
         Dim dataAdapter As New SqlDataAdapter(testCommand)

         dataAdapter.Fill(myData, "UserTable")
      End Using ' testConnection.Dispose called automatically.
      UpdateComboBox()
   End Sub
   Private Sub buttonPermData_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonPermData.Click
      ' If there is a data source, remove it.
      datagridView.DataSource = Nothing
      ' Never hard code connection strings.
      ' Usually you would get this from a config file
      Dim connectionString As String = _
       "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;"

      Using testConnection As SqlConnection = New SqlConnection(connectionString)
         Dim testCommand As SqlCommand = testConnection.CreateCommand()
         testCommand.CommandText = "Select PermissionType from PermissionsTable"
         Dim dataAdapter As New SqlDataAdapter(testCommand)

         dataAdapter.Fill(myData, "PermissionsTable")
      End Using ' testConnection.Dispose called automatically.
      UpdateComboBox()
   End Sub

   Private Sub UpdateComboBox()
      comboTables.Items.Clear()
      For Each tbl As DataTable In myData.Tables
         comboTables.Items.Add(tbl.TableName)
      Next
   End Sub
   Private Sub comboTables_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles comboTables.SelectedIndexChanged
      datagridView.DataSource = myData.Tables(comboTables.SelectedIndex)
   End Sub
End Class
